codec/src/main/java/io/netty/handler/codec/DateFormatter.java

```diff
@@ -261,40 +261,40 @@ private boolean tryParseDayOfMonth(CharSequence txt, int tokenStart, int tokenEn
         return false;
     }
 
-    private static boolean matchMonth(String month, CharSequence txt, int tokenStart) {
-        return AsciiString.regionMatchesAscii(month, true, 0, txt, tokenStart, 3);
-    }
-
     private boolean tryParseMonth(CharSequence txt, int tokenStart, int tokenEnd) {
         int len = tokenEnd - tokenStart;
 
         if (len != 3) {
             return false;
         }
 
-        if (matchMonth("Jan", txt, tokenStart)) {
+        char monthChar1 = AsciiString.toLowerCase(txt.charAt(tokenStart));
+        char monthChar2 = AsciiString.toLowerCase(txt.charAt(tokenStart + 1));
+        char monthChar3 = AsciiString.toLowerCase(txt.charAt(tokenStart + 2));
```

Comment:
Could you please try with `String#getChars(int srcBegin, int srcEnd, char dst[], int dstBegin)` with a cached `dst` please? This might help as it wouldn't count from scratch for `tokenStart + 1` and `tokenStart + 2`.
And could you please provide benchmark results with JDK8 and JDK11 too, as String implementation has change?.

Reply:
`txt` is `CharSequence`. Also, `getChars` does a copy of the array. Not sure I got what do you mean.

Reply:
@slandelle I can't =(. Can't run it due to strange errors.

> Generally speaking, I observe perf regression with JDK11 on most parsers, eg Jackson.
Same here. There were few regressions issues reported to the JDK 11. Also, the main difference - G1. G1 performs 40% worse than ParallelGC on our productions servers, so switching GC could give the large boost.